Sources/SwiftDiagnostics/GroupedDiagnostics.swift

```diff
@@ -164,11 +190,41 @@ extension GroupedDiagnostics {
 
     // If this is a nested source file, draw a box around it.
     let isRoot = sourceFile.parent == nil
-    let prefixString: String
+    var prefixString: String
     let suffixString: String
 
     if isRoot {
-      prefixString = ""
+      // If there's a primary diagnostic, print it first.
+      if let (primaryDiagSourceFile, primaryDiag) = findPrimaryDiagnostic(in: sourceFile) {
+        let primaryDiagSLC = SourceLocationConverter(fileName: primaryDiagSourceFile.displayName, tree: primaryDiagSourceFile.tree)
+        let location = primaryDiag.location(converter: primaryDiagSLC)
+
+        // Display file/line/column and diagnostic text for the primary diagnostic.
+        prefixString = "\(location.file):\(location.line):\(location.column): \(formatter.colorizeIfRequested(primaryDiag.diagMessage))\n"
+
+        // If the primary diagnostic source file is not the same as the root source file, we're pointing into a generated buffer.
+        // Provide a link back to the original source file where this generated buffer occurred, so it's easy to find if
+        // (for example) the generated buffer is no longer available.
+        if sourceFile.id != primaryDiagSourceFile.id,
```

Comment:
You can drop `.id` here. Equality of syntax nodes is defined in terms of `id`.

Reply:
Ah... this isn't `SourceFileSyntax`, but the `SourceFile` type used within grouped diagnostics.